src/client.js

```diff
@@ -67,15 +67,15 @@ function Client(manager, name, config) {
 
 	var client = this;
 
-	if (config) {
-		if (!config.token) {
-			client.updateToken(function() {
-				client.manager.updateUser(client.name, {token: config.token});
```

Comment:
The problem was specifically this line, which should have been `client.config.token`.